src/twisted/web/test/test_http.py

```diff
@@ -764,6 +768,60 @@ def _negotiatedProtocolForTransportInstance(self, t):
         return a._negotiatedProtocol
 
 
+    def test_h2CancelsH11Timeout(self):
+        """
+        When the transport is switched to H2, the HTTPChannel timeouts are
+        cancelled.
+        """
+        @attr.s
+        class FakeDelayedCall(object):
```

Comment:
Why not construct a real DelayedCall?

Also, it'd be nice if this implemented [`IDelayedCall`](https://twistedmatrix.com/documents/19.2.1/api/twisted.internet.interfaces.IDelayedCall.html) and there were a `verifyObject` call somewhere.